src/components/shared/TabItem.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div v-if="item">
+    <div v-if="withImage" class="tab-item-icon-wrapper">
+      <img class="tab-item-icon" :src="item.sprites.front_default" alt="image" />
+    </div>
+    <h1 class="tab-item-header">{{item.name}}</h1>
+    <div class="tab-item-content">
+      <div class="column" v-for="column in columns" :key="column.title">
+        <h2>{{ column.title }}</h2>
+        <div v-for="item in column.abilities" :key="item.name" class="item-wrapper">
+          <img svg-inline class="icon icon-check" src="@/assets/icons/circle-check.svg" alt="circle-check" />
+          <p>{{item[column.key].name}} <span v-if="column.key === 'stat'">: {{item.base_stat}}</span></p>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+ name: "TabItem",
```

Comment:
Same as with tabs. You want to create generic components, but sometimes it cannot be done. If the content of this component could be generic then its okay, but its more like a PokemonDescription.